### ISSUE TITLE
Node Types (+Special Nodes)

### ISSUE THREAD

**Issue by l3td33r** [juzzlin/Heimer#182]
**Nodes:**
- Text Node
- [ ] - Image Node (just a text node with a new image property?)
- [ ] - Reference Node (as above, but "reference" property?)
- [ ] - Group Node
- [ ] - Submap Node

**Text Node** - normal node, contains text.

**Image Node** - contains image and/or image+text, possibly just an extension of Text Node.

**Reference node** - displays the contents of another node in a "ghost node" that can't be edited but updates itself automatically whenever the referenced node is updated. To differentiate it from other nodes, its color can be different, for example 50% transparency. 
Special property: click on a reference move to move screen to its source node/or edit it.
Useful for large mind maps to relate nodes in different parts of the map to the same "category" and similar.

**Group Node** - you can put multiple nodes in it and it resizes automatically. When you move it, all of its contents move as well.

**Submap Node** - a node used for creating a submap/hyperlinks that reference to another heimer mindmap file. Upon click it opens the referenced mindmap. Should still keep the previous map in memory in order to return to it quickly (UI button to return to previous map?)

Concept art group/reference nodes:
![concept art heimer reference and group nodes](https://user-images.githubusercontent.com/75885440/139508272-5bf7b1b4-26a5-488e-896b-cb4f772eb30d.png)

**Comment by l3td33r** [comment]
As I think about it now, tags might be better than reference nodes for most cases, #149

**Comment by Dev5H0** [comment]
I'd like to add on to this; the ability to reference files (Full & Local paths (local being a path with the root directory being the location of the current file you're editing.)) mainly ".txt", opening them in their default editor on "ctrl+leftClick".

**Comment by oeykmiih** [comment]
this might be overblowing a bit but,

 instead of the nodes being inside an ALZ file, perhaps the ALZ only stores the connections and the actual nodes and files related are in a subfolder for that project

**Comment by oeykmiih** [comment]
as for **node group**, blender has something similar which they call "frame"

**Comment by Imerion** [comment]
Grouping nodes and getting automatic frames like that (that stay together when auto-arranging) would be really cool and useful!

**Comment by juzzlin** [comment]
Yeah, I would have needed some kind of visual grouping also myself recently. It would require some planning on how to do that properly as it should work with the layout optimizer and so on. The easiest thing to do would be something like that orange thing in the first comment and so that it won't lock the node positions, but resize automatically when the grouped nodes are individually moved. All the grouped nodes could be still moved together when dragged so that the drag point is inside the group boundaries, but not touching the nodes.
